### PR TITLE
WHOISXML API Service for Observables

### PR DESCRIPTION
**Pull request by tamthaitu**
Thank for help! I have a issues: "How to add a new analyzer? (This case is WHOISXMLAPI) #183". 
This is my pull request. 
It search successful after I edit file .env. However, another error happen. Analyzer is running and not completed. This is logs:
```
2020-09-06 17:49:40 - api_app.auth - perform_logout - INFO - perform_logout received request from root.
2020-09-06 17:49:40 - api_app.auth - perform_logout - INFO - administrator:'root' was logged out.
2020-09-06 17:49:40 - api_app.auth - perform_logout - INFO - perform_logout: current token was blacklisted.
2020-09-06 17:49:43 - api_app.auth - obtain_user_token - INFO - obtain_user_token received request for 'root'.
2020-09-06 17:49:43 - api_app.auth - obtain_user_token - INFO - headers: {'Content-Type': 'application/json', 'Content-Length': '42', 'Host': '127.0.0.1', 'User-Agent': 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:80.0) Gecko/20100101 Firefox/80.0', 'Accept': 'application/json, text/plain, */*', 'Accept-Language': 'en-US,en;q=0.5', 'Accept-Encoding': 'gzip, deflate', 'Origin': 'http://127.0.0.1', 'Connection': 'keep-alive', 'Referer': 'http://127.0.0.1/auth/login', 'Cookie': 'csrftoken=B4lrZhu61Q98kGwIQwNIRj9vTy9WkPNon3P9dptDPH8ZpKo94l7G9VP79GJqzPeO'}
2020-09-06 17:49:43 - api_app.auth - obtain_user_token - INFO - administrator:'root' was logged in.
2020-09-06 17:49:46 - api_app.api - get_analyzer_configs - INFO - get_analyzer_configs received request from root.
2020-09-06 17:49:57 - api_app.api - ask_analysis_availability - INFO - ask_analysis_availability received request from root.Data: {'md5': ['40ff44d9e619b17524bf3763204f9cbb'], 'analyzers_needed': ['Whoisxmlapi']}
2020-09-06 17:49:57 - api_app.api - send_analysis_request - INFO - send_analysis_request received request from root.Data:{'md5': '40ff44d9e619b17524bf3763204f9cbb', 'analyzers_requested': ['Whoisxmlapi'], 'run_all_available_analyzers': False, 'force_privacy': False, 'private': False, 'disable_external_analyzers': False, 'tags_id': [], 'is_sample': False, 'observable_name': '8.8.8.8', 'observable_classification': 'ip', 'runtime_configuration': {}}.
2020-09-06 17:49:57 - api_app.api - send_analysis_request - INFO - serialized_data: OrderedDict([('tags_id', []), ('is_sample', False), ('md5', '40ff44d9e619b17524bf3763204f9cbb'), ('observable_name', '8.8.8.8'), ('observable_classification', 'ip'), ('analyzers_requested', ['Whoisxmlapi']), ('run_all_available_analyzers', False), ('force_privacy', False), ('disable_external_analyzers', False), ('runtime_configuration', {})])
2020-09-06 17:49:57 - api_app.api - send_analysis_request - INFO - New Job added with ID: #1 and md5: 40ff44d9e619b17524bf3763204f9cbb.
2020-09-06 17:49:57 - api_app.script_analyzers.utils - set_job_status - INFO - setting job_id 1 to status running
2020-09-06 17:50:00 - api_app.crons - check_stuck_analysis - INFO - started check_stuck_analysis
2020-09-06 17:50:00 - api_app.script_analyzers.observable_analyzers.tor - updater - INFO - starting download of db from tor project
2020-09-06 17:50:00 - api_app.crons - check_stuck_analysis - INFO - checking if 1 jobs are stuck
2020-09-06 17:50:00 - api_app.crons - check_stuck_analysis - INFO - finished check_stuck_analysis
2020-09-06 17:50:02 - api_app.script_analyzers.observable_analyzers.tor - updater - INFO - ended download of db from tor project
```

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -9,14 +9,14 @@ INTELOWL_TAG_VERSION=v1.6.1
 
 ###### Default (Production) ######
 
-COMPOSE_FILE=docker-compose.yml
+#COMPOSE_FILE=docker-compose.yml
```
Comment (Eshaan7) [inline .env:12]: Please revert this change.

File: configuration/analyzer_config.json [hunk 2]
```diff
@@ -902,6 +902,14 @@
       "api_key_name": "VT_KEY"
     }
   },
+  "Whoisxmlapi": {
+    "type": "observable",
+    "observable_supported": ["ip", "domain"],
+    "external_service": true,
+    "description": "the WHOIS record data, of a domain name, an IP address, or an email address",
+    "requires_configuration": true,
+    "python_module": "whoisxmlapi_run"
```
Comment (Eshaan7) [inline configuration/analyzer_config.json:911]:
```
 "additional_config_params": {
      "api_key_name": "WHOISXMLAPI_KEY"
 }
```

Add this entry too just after `python_module`.

File: test.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import requests
```
Comment (Eshaan7) [inline test.py:1]: First, you shouldn't push your API key. I'd recommend that you delete this one and get a new key.
Second, the tests are to be written in [`tests/test_observables.py`](https://github.com/intelowlproject/IntelOwl/blob/master/tests/test_observables.py#L63). You can see how it's done for other analyzers and follow the same pattern. Make one entry for it under `IPAnalyzersTests` class and same entry in `DomainAnalyzersTest` class.
Third, you can delete this file.

Comment (tamthaitu) [reply test.py:1]: I will delete the test file. The purpose of this file is api test to see if it works okay, it does not involve or affect other files.

File: configuration/analyzer_config.json (hunk 2 above)
Comment (tamthaitu) [reply configuration/analyzer_config.json]: I just change and build new Intel Owl. However, result is no change. This is log:
```
2020-09-07 07:01:00 - api_app.crons - flush_expired_tokens - INFO - started flush_expired_tokens
2020-09-07 07:01:00 - api_app.crons - flush_expired_tokens - INFO - finished flush_expired_tokens
2020-09-07 07:01:25 - api_app.auth - perform_logout - INFO - perform_logout received request from root.
2020-09-07 07:01:25 - api_app.auth - perform_logout - INFO - administrator:'root' was logged out.
2020-09-07 07:01:25 - api_app.auth - perform_logout - INFO - perform_logout: current token was blacklisted.
2020-09-07 07:01:27 - api_app.auth - obtain_user_token - INFO - obtain_user_token received request for 'root'.
2020-09-07 07:01:28 - api_app.auth - obtain_user_token - INFO - headers: {'Content-Type': 'application/json', 'Content-Length': '42', 'Host': '127.0.0.1', 'User-Agent': 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:80.0) Gecko/20100101 Firefox/80.0', 'Accept': 'application/json, text/plain, */*', 'Accept-Language': 'en-US,en;q=0.5', 'Accept-Encoding': 'gzip, deflate', 'Origin': 'http://127.0.0.1', 'Connection': 'keep-alive', 'Referer': 'http://127.0.0.1/auth/login', 'Cookie': 'csrftoken=yMuneMLxwsjKfpMafVj3dvYYAKrD3TuCUWuvINYFbjIZTL2mpij3X29TmSlZaPND'}
2020-09-07 07:01:28 - api_app.auth - obtain_user_token - INFO - administrator:'root' was logged in.
2020-09-07 07:01:33 - api_app.api - get_analyzer_configs - INFO - get_analyzer_configs received request from root.
2020-09-07 07:01:43 - api_app.api - ask_analysis_availability - INFO - ask_analysis_availability received request from root.Data: {'md5': ['40ff44d9e619b17524bf3763204f9cbb'], 'analyzers_needed': ['Whoisxmlapi']}
2020-09-07 07:01:43 - api_app.api - send_analysis_request - INFO - send_analysis_request received request from root.Data:{'md5': '40ff44d9e619b17524bf3763204f9cbb', 'analyzers_requested': ['Whoisxmlapi'], 'run_all_available_analyzers': False, 'force_privacy': False, 'private': False, 'disable_external_analyzers': False, 'tags_id': [], 'is_sample': False, 'observable_name': '8.8.8.8', 'observable_classification': 'ip', 'runtime_configuration': {}}.
2020-09-07 07:01:43 - api_app.api - send_analysis_request - INFO - serialized_data: OrderedDict([('tags_id', []), ('is_sample', False), ('md5', '40ff44d9e619b17524bf3763204f9cbb'), ('observable_name', '8.8.8.8'), ('observable_classification', 'ip'), ('analyzers_requested', ['Whoisxmlapi']), ('run_all_available_analyzers', False), ('force_privacy', False), ('disable_external_analyzers', False), ('runtime_configuration', {})])
2020-09-07 07:01:43 - api_app.api - send_analysis_request - INFO - New Job added with ID: #1 and md5: 40ff44d9e619b17524bf3763204f9cbb.
2020-09-07 07:01:43 - api_app.script_analyzers.utils - set_job_status - INFO - setting job_id 1 to status running
2020-09-07 07:02:00 - api_app.crons - flush_expired_tokens - INFO - started flush_expired_tokens
2020-09-07 07:02:00 - api_app.crons - flush_expired_tokens - INFO - finished flush_expired_tokens
2020-09-07 07:03:00 - api_app.crons - flush_expired_tokens - INFO - started flush_expired_tokens
2020-09-07 07:03:00 - api_app.crons - flush_expired_tokens - INFO - finished flush_expired_tokens
2020-09-07 07:04:00 - api_app.crons - flush_expired_tokens - INFO - started flush_expired_tokens
2020-09-07 07:04:00 - api_app.crons - flush_expired_tokens - INFO - finished flush_expired_tokens
2020-09-07 07:05:00 - api_app.crons - check_stuck_analysis - INFO - started check_stuck_analysis
2020-09-07 07:05:00 - api_app.crons - flush_expired_tokens - INFO - started flush_expired_tokens
2020-09-07 07:05:00 - api_app.crons - check_stuck_analysis - INFO - checking if 1 jobs are stuck
2020-09-07 07:05:00 - api_app.crons - check_stuck_analysis - INFO - finished check_stuck_analysis
2020-09-07 07:05:00 - api_app.crons - flush_expired_tokens - INFO - finished flush_expired_tokens
```

File: .env (hunk 1 above)
Comment (tamthaitu) [reply .env]: I just change and commit update. Result is no change

File: test.py (hunk 3 above)
Comment (Eshaan7) [reply test.py]: Yes, I understood. However, we do write test cases for all analyzers like I mentioned. So instead of having a individual test file, these should be under tests/test_observables.py following the same pattern as others.

**Comment by Eshaan7** [comment]
Please follow this guide - https://github.com/wprig/wprig/wiki/How-to-squash-commits to squash all commits together.

File: test.py (hunk 3 above)
Comment (tamthaitu) [reply test.py]: What files will I have to edit in the test folder? Besides the file tests / test_observables.py, are there any other files to edit?

**Comment by tamthaitu** [comment]
> Please follow this guide - https://github.com/wprig/wprig/wiki/How-to-squash-commits to squash all commits together.
> 
> EDIT: once you have written the test cases like I mentioned, squash all commits into one. If there's any small change left after that, I'll take care of it before merging this PR. Thanks for your contribution!

**Comment by tamthaitu** [comment]
> Please follow this guide - https://github.com/wprig/wprig/wiki/How-to-squash-commits to squash all commits together.
> 
> EDIT: once you have written the test cases like I mentioned, squash all commits into one. If there's any small change left after that, I'll take care of it before merging this PR. Thanks for your contribution!

Thank for help! I think I should take some time to read more about github. I will be back tomorrow.

File: test.py (hunk 3 above)
Comment (Eshaan7) [reply test.py]: Nope, only that one.

**Comment by tamthaitu** [comment]
> > Please follow this guide - https://github.com/wprig/wprig/wiki/How-to-squash-commits to squash all commits together.
> > EDIT: once you have written the test cases like I mentioned, squash all commits into one. If there's any small change left after that, I'll take care of it before merging this PR. Thanks for your contribution!
> 
> Thank for help! I think I should take some time to read more about github. I will be back tomorrow.

**Comment by tamthaitu** [comment]
> Please follow this guide - https://github.com/wprig/wprig/wiki/How-to-squash-commits to squash all commits together.
> 
> EDIT: once you have written the test cases like I mentioned, squash all commits into one. If there's any small change left after that, I'll take care of it before merging this PR. Thanks for your contribution!

Now I squash all commit in branch `develop`  is up to date with the `master` branch?

**Comment by Eshaan7** [comment]
> Now I squash all commit in branch develop is up to date with the master branch?

No, just follow the guide I mentioned. You just need to squash the last 6 commits you made into a single one.

**Comment by tamthaitu** [comment]
> > Now I squash all commit in branch develop is up to date with the master branch?
> 
> No, just follow the guide I mentioned. You just need to squash the last 6 commits you made into a single one. If you are having trouble doing so, that's okay, let me know and I can do it for you.

Thank for help! I just squash commit in the develop branch by follow the guide you mentioned. Hope you help me add new analyzer successful.

File: intel_owl/tasks.py [hunk 4]
```diff
@@ -55,6 +55,7 @@
     tranco,
     pulsedive,
     intelx,
+    whoisxmlapi,
```
Comment (Eshaan7) [inline intel_owl/tasks.py:58]: The error was because you forgot to import this here.
